### ISSUE TITLE
[NETE2ESDK] [Linux] Downloading .NET failed with Error:(TimeoutSudoProcessSpawnerError)

### ISSUE THREAD

**Issue by v-xiaofchen** [dotnet/vscode-dotnet-runtime#1995]
### Describe the bug.

Downloading .NET failed with Error:(TimeoutSudoProcessSpawnerError)

### Please attach log file(s) here if you have any.

The log file location should be copied to your clipboard if you came from VS Code.

### Reproduction Steps

1. On Ubuntu 22.04 or RHEL 9.3, run the sample command ‘Install .NET SDK Globally via .NET Install Tool (Former Runtime Extension)‘.
2. Use the version it provided, which is 7.0.103. Hit enter.
3. Observe

### Expected Behavior

Download .NET successfully

### Exceptions & Errors (if any)

Downloading .NET failed with Error:(TimeoutSudoProcessSpawnerError)
![Image](https://github.com/user-attachments/assets/dc94f04c-f782-43c1-b024-c95e8cb62792)

### .NET Install Tool Version

2.2.1

### Your operating system? Please provide more detail, such as the version and or distro above.

N/A

### VS Code Version

1.94.2

**Comment by nagilson** [comment]
@v-xiaofchen Did you run anything on this machine for these failures before-hand besides the 'Install SDK' command a single time?
I could not get this to repro for me. 

--

Investigation:
`ok.txt` was written even though it never should have existed. It locks and wipes the directory before-hand, nobody else writes `ok.txt` so I dont understand why it said it was there already. `fs.readFileSync(filePath).toString()` will fail if the file DNE, so it was definitely there, not that it returned '' and then thought ok.txt existed already. I updated the logging in the build to see if wiping the dir worked and let's see if it was created earlier.

Also, we never awaited the sudo process startup and I dont think we should need to. But it never got past the phase of trying to load the process. Im not sure why that's a problem. I assume the user hit the password prompt. Not sure if there are logs generated to tell. I should provide a build with `await` and see if that helps

**Comment by nagilson** [comment]
It looks like this is not an issue with the code. The issue only repros all of the time, and it only repros exclusively on the signed package vsix builds. It does not repro with the same exact code build which is not the extra signed build from the pipeline.

The vscode sudo prompt library hangs indefinitely and never fails, exits, etc and hangs on the build with `-signed`.  This may be due to some changes around signing vscode made in October.

**Comment by v-xiaofchen** [comment]
Yes, this issue repro on the signed build vscode-dotnet-runtime-2.2.2-signed
![Image](https://github.com/user-attachments/assets/eb2fac00-0910-44c3-b4ba-0d40c2e4f414)
But it does not reproduce on the one without -signed version vscode-dotnet-runtime-2.2.2
![Image](https://github.com/user-attachments/assets/d843d848-72c9-4500-989e-ae07ae83851f)

**Comment by v-xiaofchen** [comment]
This issue does not repro on vscode-dotnet-runtime-2.2.3.
![Image](https://github.com/user-attachments/assets/1126001c-386d-48c0-918c-da1c8afd98c4)

**Comment by v-xiaofchen** [comment]
This issue does not repro on the new vscode-dotnet-runtime-2.2.3.
![Image](https://github.com/user-attachments/assets/faa41900-9686-47da-ac3e-8950cd7b6597)

**Comment by YingyingYuan9** [comment]
This issue is not reproduced on the new build 2.3.0

![Image](https://github.com/user-attachments/assets/12cd4a4f-f100-4e51-9424-151d0a404a72)
